### PR TITLE
Brelee/elmo branchy boi

### PR DESCRIPTION
**Pull request by Brelee2222**

### CODE REVIEW

**Comment by rutmanz** [comment]
I'm going to wait for Micah's PR to be merged before I review this, the git history will be confusing otherwise

**Comment by Brelee2222** [comment]
ok

**Comment by rutmanz** [comment]
#4 is merged now, can you update your branch?

**Comment by Brelee2222** [comment]
> #4 is merged now, can you update your branch?

is what I did right?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (rutmanz) [inline .vscode/settings.json]: The .vscode folder probably shouldn't be committed

File: build.gradle [hunk 2]
```diff
@@ -1,15 +1,12 @@
-import edu.wpi.first.gradlerio.deploy.roborio.RoboRIO
-
 plugins {
```
Comment (rutmanz) [inline build.gradle]: I'm going to assume that this file was automatically generated, but the old version crucially had the correct main class. You won't be able to build or deploy with this gradle configuration

File: vendordeps/Phoenix.json [hunk 3]
```diff
@@ -0,0 +1,257 @@
+{
```
Comment (rutmanz) [inline vendordeps/Phoenix.json]: This dependency is already here under a different name, don't need it again

**Review by rutmanz — COMMENTED** [review]
Your drivetrain and tank drive classes look good. The other files are all either already there from Micah's PR (gradle files and phoenix vendordep) or don't need to be commited (.vscode)

**Comment by rutmanz** [comment]
I'd also like to emphasize that having commit messages that actually describe your changes is important so that other people can see why each commit is there

**Comment by Brelee2222** [comment]
> I'd also like to emphasize that having commit messages that actually describe your changes is important so that other people can see why each commit is there

Sorry. I should probably start making an effort to do that

File: .vscode/settings.json (hunk 1 above)
Comment (Brelee2222) [reply .vscode/settings.json]: ok

File: vendordeps/Phoenix.json (hunk 3 above)
Comment (Brelee2222) [reply vendordeps/Phoenix.json]: I'll delete it

**Comment by rutmanz** [comment]
I merged all but the last commit of this manually. You can make a new pull request for the apriltag stuff

**Comment by rutmanz** [comment]
Looks like it's not updated to use the correct constants, I'll unmerge it for now and we'll test without it

**Comment by Brelee2222** [comment]
> Looks like it's not updated to use the correct constants, I'll unmerge it for now and we'll test without it

I don't know the constants for the motors

**Review by rutmanz — APPROVED** [review]

**Pull request merged**